pkg/distributors/test/MerkleRedeem.test.ts

```diff
@@ -0,0 +1,273 @@
+        rewardTokens,
+        [{ account: lp2, changes: { DAI: claimedBalance } }],
+        vault
+      );
+
+      const isClaimed = await merkleRedeem.claimed(1, lp2.address);
+      expect(isClaimed).to.equal(true); // "claim should be marked as claimed";
+    });
+
+    it("Doesn't allow a user to claim for another user", async () => {
+      const claimedBalance = bn('1000');
+      const merkleProof = merkleTree.getHexProof(elements[0]);
+
+      const errorMsg = 'Incorrect merkle proof';
+      expect(
+        merkleRedeem.connect(lp3).claimWeek(lp3.address, 1, claimedBalance, merkleProof, false)
+      ).to.be.revertedWith(errorMsg);
+    });
+
+    it('Reverts when the user attempts to claim the wrong balance', async () => {
+      const claimedBalance = bn('666');
+      const merkleProof = merkleTree.getHexProof(elements[0]);
+      const errorMsg = 'Incorrect merkle proof';
+      expect(
+        merkleRedeem.connect(lp2).claimWeek(lp2.address, 1, claimedBalance, merkleProof, false)
+      ).to.be.revertedWith(errorMsg);
+    });
+
+    it('Reverts when the user attempts to claim twice', async () => {
+      const claimedBalance = bn('1000');
+      const merkleProof = merkleTree.getHexProof(elements[0]);
+
+      await merkleRedeem.connect(lp2).claimWeek(lp2.address, 1, claimedBalance, merkleProof, false);
+
+      const errorMsg = 'cannot claim twice';
+      expect(
+        merkleRedeem.connect(lp2).claimWeek(lp2.address, 1, claimedBalance, merkleProof, false)
+      ).to.be.revertedWith(errorMsg);
+    });
+  });
+
+  describe('When a user has several allocation to claim', () => {
+    const claimBalance1 = bn('1000');
+    const claimBalance2 = bn('1234');
+
+    let elements1: string[];
+    let merkleTree1: MerkleTree;
+    let root1: string;
+
+    let elements2: string[];
+    let merkleTree2: MerkleTree;
+    let root2: string;
+
+    beforeEach(async () => {
+      elements1 = [encodeElement(lp2.address, claimBalance1)];
+      merkleTree1 = new MerkleTree(elements1);
+      root1 = merkleTree1.getHexRoot();
+
+      elements2 = [encodeElement(lp2.address, claimBalance2)];
+      merkleTree2 = new MerkleTree(elements2);
+      root2 = merkleTree2.getHexRoot();
+
+      await merkleRedeem.seedAllocations(bn(1), root1, claimBalance1);
+
+      await merkleRedeem.seedAllocations(bn(2), root2, claimBalance2);
+    });
+
+    it('Allows the user to claim multiple weeks at once', async () => {
+      const claimedBalance1 = bn('1000');
+      const claimedBalance2 = bn('1234');
+
+      const proof1: BytesLike[] = merkleTree1.getHexProof(elements1[0]);
+      const proof2: BytesLike[] = merkleTree2.getHexProof(elements2[0]);
+
+      const merkleProofs = [
+        { week: bn(1), balance: claimedBalance1, merkleProof: proof1 },
+        { week: bn(2), balance: claimedBalance2, merkleProof: proof2 },
+      ];
+
+      await merkleRedeem.connect(lp2).claimWeeks(lp2.address, merkleProofs, false);
```

Comment:
Let's also add `expectBalanceChange` and event tests here (maybe in different `it`s).